### ISSUE TITLE
can detect more than seven numbers?

### ISSUE THREAD

**Issue by willduan** [matthewearl/deep-anpr#16]
i'm very appreciate to know the project? I just want to use to detect some numbers and chracters? can it detect more than seven numbers?  Thanks

**Comment by sapeyes** [comment]
I am not sure about the performance. As a glance, that is possible. 
By adding more digits (common.py), the artificially made training images would have more digits.
The network used in here should be extended to accept more digits. 
The easiest way is to increase the dimension of last FC output.

**Comment by sourabh2k15** [comment]
Yes ofcourse it can, but the code modification doesn't involve changing common.py , but you have to change gen.py , the generate code function

**Comment by ghost** [comment]
@sourabh2k15 I've tried that, and I don't believe it works. I think you do have to modify code elsewhere other than gen.py because I believe especially the output of the fully connected layer thinks there are only 7 classes. It would be great if anyone figures an easy way to change the number of characters the nn expects.

**Comment by sourabh2k15** [comment]
I had done it successfully, Indian license plates have 10 length. so i
changed in gen.py the pattern , got test data , then changed model.py to
change 7 to 10 everywhere, you also need to change it in train.py . Also
finally when you get training weights you will want to run it on new images
using detect.py, there also change 7s to 10s . Hope that helps.

Thanks.

**Comment by sourabh2k15** [comment]
I had done it successfully, Indian license plates have 10 length. so i changed in gen.py the pattern , got test data , then changed model.py to change 7 to 10 everywhere, you also need to change it in train.py . Also finally when you get training weights you will want to run it on new images using detect.py, there also change 7s to 10s . Hope that helps.

Thanks.
